### PR TITLE
Display clouds integration test

### PR DESCRIPTION
**Pull request by SimonRichardson**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

The following changes introduce the idea of a display clouds integration
tests. Unfortunately it required to change how we handle machine
interactions. For formatting directives such as json or yaml, we never
want to be prompted for anything, we just either want an empty output in
the right format (json being {}) or the data.

Bringing in the changes is the first drive into doing so, the PR doesn't
enforce the changes through out, although it is a first step in doing
so.

Adding redirection of the stderr to `/dev/null` for machine directives is
pointless, so in the future checking for `ctx.IsSerial()` should be used.
If the value is true then omitting the information and the noise is
probably the wisest.

**See:** https://discourse.jujucharms.com/t/cli-serialisable-output-formats/2490

## QA steps

*Please replace with how we can verify that the change works?*

```sh
cd tests && ./main.sh cli test_display_clouds
```

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/Y9Kj7cIi/2152-move-display-cloud-to-integration-tests

File: tests/suites/cli/display_clouds.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  ],
+  "endpoint": "http://192.168.1.21/MAAS",
+  "regions": {
+    "default": {}
+  }
+}
+EOF
+  )
+
+  OUT=$(XDG_DATA_HOME="${TEST_DIR}" juju clouds --all --format=json 2>/dev/null | jq ".[] | select(.defined != \"built-in\")")
```
[the remark this answers is not in the document]
Comment (SimonRichardson) [reply tests/suites/cli/display_clouds.sh]: Running: `./main.sh -V cli test_desplay_clouds` (notice capital `V`) will tell you exactly what's running... It's easier to see what's going on.

File: tests/suites/cli/display_clouds.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  echo "" >> "${TEST_DIR}/juju/credentials.yaml"
+
+  OUT=$(XDG_DATA_HOME="${TEST_DIR}" juju clouds --local --format=json 2>/dev/null | jq ".[] | select(.defined != \"built-in\")")
+  if [ -n "${OUT}" ]; then
+    echo "expected empty ${OUT}"
+    exit 1
+  fi
+
+  cp ./tests/suites/cli/clouds/public-clouds.yaml "${TEST_DIR}"/juju/public-clouds.yaml
+  OUT=$(XDG_DATA_HOME="${TEST_DIR}" juju clouds --local --format=json 2>/dev/null | jq ".[] | select(.defined != \"built-in\")")
```
[the remark this answers is not in the document]
Comment (SimonRichardson) [reply tests/suites/cli/display_clouds.sh]: Running: ./main.sh -V cli test_desplay_clouds (notice capital V) will tell you exactly what's running... It's easier to see what's going on.

**Comment by SimonRichardson** [comment]
!!build!!

File: tests/suites/cli/display_clouds.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (hmlanigan) [reply tests/suites/cli/display_clouds.sh]: Okay that worked, but not exactly human reading friendly.

[the remark this answers is not in the document]
Comment (SimonRichardson) [reply tests/suites/cli/display_clouds.sh]: I wonder if we can do something about that and only trace out certain things, let me think about this.

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

> FAIL: caasoperator_test.go:437: WorkerSuite.TestContainerStart

**Pull request merged**